docs/_future/deploy_your_app.md

```diff
@@ -7,10 +7,249 @@ layout: tutorial
+
+Since we're deploying to Heroku, having a Heroku account will also be useful. If you don't have one, [create one here](https://signup.heroku.com/).
+
+In order to manage, deploy, and debug your app on Heroku, you'll also need to [install the Heroku CLI](https://devcenter.heroku.com/articles/getting-started-with-nodejs#set-up) onto your machine and authenticate using `heroku login`.
+
+And lastly, to manage the version of your project being deployed, you'll need to [install Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and initialize your app's repository using `git init` if you have not already done so.
+
+---
+
+### Create a Heroku app {#create-a-heroku-app}
+
+With a Bolt app ready to go and Heroku tools installed, you're now ready to start deploying your Bolt app on Heroku!
+
+#### Create a new app
+
+From the command line, go to your project directory and start by creating a Heroku app with a unique name:
+
+```sh
+$ heroku create your-heroku-app
+```
+
+or have some fun with a random name:
+
+```sh
+$ heroku create
+# Creating app... done, ⬢ aqueous-escarpment-85734
+# https://aqueous-escarpment-85734.herokuapp.com/ | https://git.heroku.com/aqueous-escarpment-85734.git
+```
+
+After your Heroku app is created, you'll be given app-specific information for use in later steps. From the example above:
+
+* App name: `aqueous-escarpment-85734`
+* Web address: `https://aqueous-escarpment-85734.herokuapp.com/`
+* Git remote: `https://git.heroku.com/aqueous-escarpment-85734.git`
+
+#### Add environment variables
+
+The tokens being used for your app (and any other environment variables) can be added to your deployed environment using the following commands:
+
+```sh
+$ heroku config:set SLACK_APP_TOKEN=xapp-your-app-level-token
+$ heroku config:set SLACK_BOT_TOKEN=xoxb-your-bot-token
+```
+
+These tokens can be collected from the "Basic Information" and "OAuth & Permissions" sections of [your App Config page](https://api.slack.com/apps) if ever misplaced.
+
+Note that these tokens and variables don't necessarily have to match those used in development! For instance, you may want to use tokens for a different app or a more verbose logging output in production. This is where you would set that.
+
+### Prepare your app for Heroku {#prepare-your-app}
+
+For a successful deployment, you'll need to set up a few additional things in your app.
+
+#### Add a Procfile {#procfile}
+
+To specify the start command of your app for Heroku, [a special file called `Procfile`](https://devcenter.heroku.com/articles/procfile) must be created in your app's root directory.
+
+The contents of this file will vary depending on if you have [Socket Mode](/bolt-js/concepts#socket-mode) enabled or not (which you can check for in your [Manifest](/bolt-js/future/app-manifests)) since apps using default web connections use HTTP requests when communicating with Slack.
```

Comment:
Feel free to adjust the formatting / verbiage of this one as you see fit! Also updated the socket mode link to a better reference but lmk your thoughts on if you feel the bolt JS guide for Socket Mode is a better fit here!

```suggestion
The contents of this file will vary depending on if you have [Socket Mode](https://api.slack.com/apis/connections/socket) enabled or not. Socket Mode allows for your app to receive payloads through a private [WebSocket](https://tools.ietf.org/html/rfc6455) URL that's generated at runtime rather than a public HTTP endpoint. To confirm whether Socket Mode is enabled with your application, you can either:
* Check your app's [Manifest](/bolt-js/future/app-manifests) to see if the `socketModeEnabled` property has been added and equals `true`, or
* Visit the [App Config page](https://api.slack.com/apps) for your application and check out the "Socket Mode" section to ensure the toggle is turned on:
<img width="670" alt="Screen Shot 2022-10-25 at 3 49 42 PM" src="https://user-images.githubusercontent.com/12901850/197868321-63fff839-a3a2-4926-ae45-93c9d88cfd41.png">
```

Reply:
That api.slack.com page explains the Socket Mode concept quite nicely! I like this change since this seems to answer what Socket Mode is without subtly suggesting that this is the better way to run a Bolt app (which may be implied by linking to related docs that show _how_ to connect using sockets) 😄